---
title: "ESR1 bulk calling card analysis following on from SP1 analysis"
author: "Jack Stenning"
date: "2023-11-28"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Background

## Rationale

After running the overlap analysis on SP1 calling cards and SP1 ChIP-Seq, I wanted to try the same analysis on my own ESR1 bulk calling cards and an existing ChIP-Seq data set - GSE109820. The processing of ESR1 calling cards was carried out by following the protocol.io pages:

'Processing Bulk Calling Card Sequencing Data' <https://www.protocols.io/view/processing-bulk-calling-card-sequencing-data-n2bvjr25xlk5/v1> 'Calling Peaks on piggyBac Calling Card Data' <https://www.protocols.io/view/calling-peaks-on-piggybac-calling-card-data-4r3l289dql1y/v1>

The Rmd file that generated this document is executable. The data required for this analysis can be found in the '202306_JS_BulkCC' google drive link shared with you. To save time, all of the code blocks up  to 'Working in R studio' have already been run and their outputs were included in the '4 - Window outputs to plot in R' directory. Please make sure this directory is in the same location as the Rmd file before beginning.

Additional data files such as the original called peak files, qbed and fastq files are also available from the google drive link in folders that begin with '3, 2 and 1' respectively if you wish to replicate my analysis.

# Method

## Working on the Univeristy of York Viking HPCC

### Prepare files for peak calling
#### Data found in '1 - 202306_JS_BulkCC'

Firstly, the barcode and index sequence had to be removed from the raw reads ER and WT calling cards. Samples were differentiated by using two different barcodes to denote whether the calling cards were directed by ER through the HyPB-ER fusion protein, or whether they are undirected by the Wild Type HyPB protein. Each replicate carried a unique Nextera index sequence to identify replicates within conditions.

Given that all of the barcodes are the same for samples within each condition, a loop script was written to trim the barcodes from ER directed and undirected calling cards.

```{bash cut barcode, eval=FALSE}
#Load modules
module load cutadapt/3.4-GCCcore-10.3.0


#First step is to cut off the barcode and surrounding adapter
#For ER calling cards in this experiment, the barcode is the same - CGT

cd 202306_JS_BulkCC/
for direct in ER*; do
cd "$direct"
gunzip *.gz
	for file in *1.fq; do
		cutadapt \
		    -g ^CGTTTTACGCAGACTATCTTTCTAGGGTTAA \
		    --minimum-length 1 \
		    --discard-untrimmed \
		    -e 0 \
		    --no-indels \
	 	   -o "$file"TrimmedBC.fq \
	 	   "$file"
	done
	cd ../
done
cd ../

#For WT calling cards in this experiment, the barcode is the same - TAG

cd 202306_JS_BulkCC/
for direct in WT*; do
cd "$direct"
gunzip *.gz
	for file in *1.fq; do
		cutadapt \
		    -g ^TAGTTTACGCAGACTATCTTTCTAGGGTTAA \
		    --minimum-length 1 \
		    --discard-untrimmed \
		    -e 0 \
		    --no-indels \
	 	   -o "$file"TrimmedBC.fq \
	 	   "$file"
	done
	cd ../
done
cd ../

```

However, although the index sequence is different for each of the replicates within a condition, trimming of these sequences can be carried out in one script by replacing the index sequence with 'NNNNNNNNNN' as cut adapt can handle degenerate sequence within adaptors. The code below is an example of the processing for a single replicate - ER1, other replicates use the same code, with the file name and index sequence supplemented for the correct versions.

```{bash cut index, eval=FALSE}
#Load modules
module load cutadapt/3.4-GCCcore-10.3.0

#Cut adapt will take degenerate sequence for indexes - NNNNNNNNNN
cd 202306_JS_BulkCC/
for dir in *; do
  cd "$dir"
  echo "Now in "$dir""
  for file in *TrimmedBC.fq; do
  	cutadapt \
  	-a CTGTCTCTTATACACATCTCCGAGCCCACGAGACTNNNNNNNNNNTCTCGTATGCCGTCTTCTGCTTG \
  	--minimum-length 1 \
  	-o "$file"_fulltrimmed.fastq \
  	"$file"
  done
  cd ../
done
cd ../

```

Once all of the files have been trimmed, the reads can be aligned to hg38. Contrary to the protocol.io, BWA MEM was used as novoalign was not available on the Viking HPCC. The code below is an example of the processing for a single replicate - ER1, other replicates use the same code, with the file name supplemented for the correct versions.

```{bash align, eval=FALSE}
#purge then load modules
module purge
module load deepTools/3.5.0-foss-2021a
module load BWA/0.7.17-foss-2019b
module load SAMtools/1.16.1-GCC-11.3.0
module load BEDTools/2.30.0-GCC-11.2.0

cd 202306_JS_BulkCC/
for dir in *; do
  cd "$dir"
  echo "Now in "$dir""
  for file in *fulltrimmed.fastq; do
   	echo "Processing "$file""
	  bwa mem -p -t 40 /mnt/scratch/users/jps558/staging/hg38/hg38.fa "$file" > "$file".sam
	  done
	  cd ../
done
cd ../
	  
	  
```

After, the reads can be sorted into a .bam file.

```{bash sort to bam, eval=FALSE}
#Purge then load modules
		module purge
		module load SAMtools/1.16.1-GCC-11.3.0
		
#Sorting to bam
cd 202306_JS_BulkCC/
for dir in *; do
  cd "$dir"
  echo "now in "$dir""
	for file in *.sam; do
	  echo "$file"
		samtools view \
 	  	 	-bS -h -F 260 \
 	  		 "$file" | \
 	   		samtools sort - -o "$file"_mapped.bam
		done
    cd ../
done
cd ../
```

After this step, the TagBam.py script from Arnhav Moudgil's GitHub - <https://github.com/arnavm/calling_cards> -is run. Similarly to the adaptor trimming step, given that all of the barcodes are the same for samples within each condition, a loop script was written to trim the barcodes from ER directed and undirected calling cards. However, as the index sequence is different for each of the replicates within a condition, trimming of these sequences must be done individually for each replicate.

In order to execute python code on the Viking2 HPCC I first had to set up a virtual environment.
```{bash set up env, eval=FALSE}
### Set up env
module load Miniconda3/23.5.2-0
conda env create -f CallingCardsPython.yaml
cd ../
```


```{bash tag barcodes, eval=FALSE}
#Purge then load modules
module purge
module load SAMtools/1.16.1-GCC-11.3.0
module load BEDTools/2.30.0-GCC-11.2.0

#Activate python environment
module load Miniconda3/23.5.2-0
#Activate python environment

source activate CallingCardsPython

#Tag barcodes
cd 202306_JS_BulkCC/
for direct in ER*; do
cd "$direct"
	for file in *mapped.bam; do
		python ../../TagBam.py \
 		   --tag XP:Z:CGT \
 		   "$file" \
 		   "$file"_tagged.bam	
	done
	cd ../
done
for direct in WT*; do
cd "$direct"
	for file in *mapped.bam; do
		python ../../TagBam.py \
 		   --tag XP:Z:TAG \
 		   "$file" \
 		   "$file"_tagged.bam	
	done
	cd ../
done
cd ../

#TAG WT barcodes
echo "bash TAG WT barcodes"
cd 202306_JS_BulkCC/
for direct in WT*; do
cd "$direct"
	for file in *mapped.bam; do
		python ../../TagBam.py \
 		   --tag XP:Z:TAG \
 		   "$file" \
 		   "$file"_tagged.bam	
	done
	cd ../
done
cd ../


```

Once the barcodes are tagged, the index sequence can also be tagged.

```{bash tag index, eval=FALSE}
echo "bash tag index"
cd 202306_JS_BulkCC/
#ER1
cd ER1/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:ACTCGCTA \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#ER2
cd ER2/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:GGAGCTAC \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#ER3
cd ER3/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:GCGTAGTA \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#ER5
cd ER5/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:CGGAGCCT \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#ER6
cd ER6/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:TACGCTGC \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
################################################################################################
#WT1
cd WT1/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:ATGCGCAG \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#WT2
cd WT2/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:TAGCGCTC \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#WT3
cd WT3/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:ACTGAGCG \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#WT4
cd WT4/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:CCTAAGAC \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#WT5
cd WT5/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:CGATCAGT \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
#WT6
cd WT6/
for file in *_tagged.bam; do
	python ../../TagBam.py \
	   --tag XJ:Z:TGCAGCTA \
	   "$file" \
	   "$file"_tagged2.bam	
done
cd ../
```

Once all of the replicates have had their index tag added, the insertion sites can be annotated using the AnnotateInsertionSites.py script. After the .bam files can be indexed

```{bash annotate insert, eval=FALSE}
echo "bash annotate insert"
#Purge then load modules
module purge
module load SAMtools/1.16.1-GCC-11.3.0
module load BEDTools/2.30.0-GCC-11.2.0

#Prepare python environment
module load Miniconda3/23.5.2-0
source activate CallingCardsPython

cd 202306_JS_BulkCC
for dir in *; do
	cd "$dir"
	echo "$dir"
	for file in *tagged2.bam; do
		python ../../AnnotateInsertionSites.py \
		    --transposase PB \
		    -f \
		    "$file" \
		    /mnt/scratch/users/jps558/staging/hg38/hg38.2bit \
 		   "$file"_final.bam
		done
	for mile in *mapped.bam_tagged.bam_tagged2.bam_final.bam; do
            mv "$mile" "${mile%%mapped.bam_tagged.bam_tagged2.bam_final.bam}final.bam"
        done
	cd  ../
done
cd ../
```

```{bash index bam, eval=FALSE}
echo "bash index bam"
module purge
module load SAMtools/1.16.1-GCC-11.3.0

cd 202306_JS_BulkCC/
for dir in *; do
	cd "$dir"
	echo "$dir"
	for file in *final.bam; do
		samtools index "$file"
		done
	cd  ../
done
cd ../

```

Finally, the files can be cleaned up and concatenated into a single file for easier use down the pipeline.

```{bash convert Bam to CC, eval=FALSE}
echo "bash convert bam to CC"
module purge
module load SAMtools/1.16.1-GCC-11.3.0
module load BEDTools/2.30.0-GCC-11.2.0

module load Miniconda3/23.5.2-0
source activate CallingCardsPython

cd 202306_JS_BulkCC/
mkdir outputs
mkdir outputs/ER
mkdir outputs/WT
for dir in ER*; do
	cd "$dir"
	for file in *final.bam; do
		python ../../BamToCallingCard.py \
		    -b XP XJ \
		    -i "$file" \
		    > "$file".qbed
		done
	for mile in *final.bam.qbed; do
            mv "$mile" "${mile%%final.bam.qbed}.qbed"
        done
  for save in *.qbed; do
    cp *.qbed ../outputs/ER
    done
  for store in *final.bam*; do
    cp "$store" ../outputs/ER
    done
	cd  ../
done

for dir in WT*; do
	cd "$dir"
	for file in *final.bam; do
		python ../../BamToCallingCard.py \
		    -b XP XJ \
		    -i "$file" \
		    > "$file".qbed
		done
	for mile in *final.bam.qbed; do
            mv "$mile" "${mile%%final.bam.qbed}.qbed"
        done
  for save in *.qbed; do
    cp *.qbed ../outputs/WT
    done
  for store in *final.bam*; do
    cp "$store" ../outputs/WT
    done
	cd  ../
done

#Cat ER
cd outputs/ER
mkdir combined/
cat ER*HF552DSX7_L2_1_.qbed | bedtools sort -i > combined/ER_HyPB_HF552DSX7_L2_1.qbed
cat ER*HFK3FDSX7_L2_1_.qbed | bedtools sort -i > combined/ER_HyPB_HFK3FDSX7_L2_1.qbed
cat ER*.qbed | bedtools sort -i > combined/ER_HyPB_EKDL230008858.qbed
samtools merge ER_HyPB_EKDL230008858_final.bam ER*final.bam
cd combined/
mv *qbed ../
cd ../../

#Cat WT
cd WT
mkdir combined/
cat WT*HF552DSX7_L2_1_.qbed | bedtools sort -i > combined/WT_HyPB_HF552DSX7_L2_1.qbed
cat WT*HFK3FDSX7_L2_1_.qbed | bedtools sort -i > combined/WT_HyPB_HFK3FDSX7_L2_1.qbed
cat WT*.qbed | bedtools sort -i > combined/WT_HyPB_EKDL230008858.qbed
samtools merge WT_HyPB_EKDL230008858_final.bam WT*final.bam
cd combined/
mv *qbed ../
cd ../../../../


```

You can also clean up intermediary file types if you dont remove the files you are interested in

```{bash cleanup, eval=FALSE}
for dir in 202306_JS_BulkCC/*; do
	cd "$dir"
	for file in *Trimmed*; do
		rm "$file"
		done
	for dile in *.sam; do
		rm "$dile"
		done
	for mile in tagged*.bam; do
		rm "$mile"
		done
	cd ../
	done
cd ../

```

### Peak calling
#### Data found in '2 - .qBED files needed from Peak Calling'

Before calling peaks, it's necessary to generate a list of 'TTAA' tetranucleotides throughout hg38. This is used to generate an estimate for the expected uniform insertion rate of calling cards because the HyPB transposase inserts almost exclusively at these sites. This can be done using the kmer program. There is an optional step that can be taken to use only canonical chromosomes.

```{bash get TTAA, eval=FALSE}
chmod +x kmer
./kmer ../staging/hg38/hg38.fa TTAA > ../staging/hg38/hg38_TTAA.bed

grep -v '_' ../staging/hg38/hg38_TTAA.bed > ../staging/hg38/hg38_TTAA_canon.bed
```

the qBED files should then be sorted before processing further.

```{bash sort qBED, eval=FALSE}
module load BEDTools/2.30.0-GCC-11.2.0
cd 202306_JS_BulkCC/outputs
#ER
cd ER
bedtools sort -i ER_HyPB_HF552DSX7_L2_1.qbed > ER_HyPB_HF552DSX7_L2_1_sorted.qbed
bedtools sort -i ER_HyPB_HFK3FDSX7_L2_1.qbed > ER_HyPB_HFK3FDSX7_L2_1_sorted.qbed
bedtools sort -i ER_HyPB_EKDL230008858.qbed > ER_HyPB_EKDL230008858_sorted.qbed
mkdir notsorted
cp *1.qbed notsorted/
cp *8.qbed notsorted/
cd ../
#WT
cd WT
bedtools sort -i WT_HyPB_HF552DSX7_L2_1.qbed > WT_HyPB_HF552DSX7_L2_1_sorted.qbed
bedtools sort -i WT_HyPB_HFK3FDSX7_L2_1.qbed > WT_HyPB_HFK3FDSX7_L2_1_sorted.qbed
bedtools sort -i WT_HyPB_EKDL230008858.qbed > WT_HyPB_EKDL230008858_sorted.qbed
mkdir notsorted
cp *1.qbed notsorted/
cp *8.qbed notsorted/
cd ../../../
```

I had previously filtered blacklisted sites from SP1 calling cards before creating block files, I will do the same here but will also process the unfiltered reads alongside.

```{bash filter, eval=FALSE}
#Load modules
module load BEDTools/2.30.0-GCC-11.2.0

cd 202306_JS_BulkCC/outputs/ER

#Filter
for file in *sorted.qbed; do
    bedtools subtract -a  "$file" -b /mnt/scratch/users/jps558/staging/hg38/Blacklist/hg38-blacklist.v2.bed > \
    "$file"_filtered.qbed
  done
cd ../WT
for file in *sorted.qbed; do
    bedtools subtract -a  "$file" -b /mnt/scratch/users/jps558/staging/hg38/Blacklist/hg38-blacklist.v2.bed > \
    "$file"_filtered.qbed
  done
cd /mnt/scratch/users/jps558/Stenning_Data_Analysis
```


Then, the sorted files can be used to generate the blocks file, this segments the reads into regions that contain a uniform insertion rate. This is used by later scripts to determine which insertions are background, and which are genuine re-directed insertions. Finally, the files can be used to call peaks, and there are separate scripts used for directed and un-directed calling cards. 

Additional script has been included here to clean the file ends of unnecessarily long suffix'.

```{bash create blocks and call peaks, eval=FALSE}
module purge
module load SAMtools/1.16.1-GCC-11.3.0
module load BEDTools/2.30.0-GCC-11.2.0
module load Miniconda3/23.5.2-0


#Activate python environment


source activate CallingCardsPython



cd 202306_JS_BulkCC/outputs/ER
for file in *filtered.qbed; do
  echo "$file"
  #python /mnt/scratch/users/jps558/Stenning_Data_Analysis/SegmentCCF.py "$file" | sed -e '/^\s*$/d' > "$file".blocks
  python /mnt/scratch/users/jps558/Stenning_Data_Analysis/BBPeakCaller_TF.py -a 0.05 -m fdr_bh -d 250 -x 5000 \
    -i "$file"_no_r_intermediate.csv \
  	"$file" \
	  "$file".blocks \
	  ../WT/WT_HyPB_EKDL230008858_sorted.qbed \
  	"$file"_fdr_bh_0.05_peaks_nor.bed
  python /mnt/scratch/users/jps558/Stenning_Data_Analysis/BBPeakCaller_TF.py -a 0.05 -r -m fdr_bh -d 250 -x 5000 \
	  -i "$file"_r_intermediate.csv \
  	"$file" \
	  "$file".blocks \
	  ../WT/WT_HyPB_EKDL230008858_sorted.qbed \
  	"$file"_fdr_bh_0.05_peaks_r.bed
  done
cd ../WT
for file in *filtered.qbed; do
  echo "$file"
  #python /mnt/scratch/users/jps558/Stenning_Data_Analysis/SegmentCCF.py "$file" | sed -e '/^\s*$/d' > "$file".blocks
  python /mnt/scratch/users/jps558/Stenning_Data_Analysis/BBPeakCaller_BRD4.py -p 30 -d 12500 \
    -i "$file"_p30_intermediate.csv \
    "$file" \
  	"$file".blocks \
  	/mnt/scratch/users/jps558/staging/hg38/hg38_TTAA.bed \
  	"$file"_p30_peaks.bed
  python /mnt/scratch/users/jps558/Stenning_Data_Analysis/BBPeakCaller_BRD4.py -p 62 -d 12500 \
    -i "$file"_p62_intermediate.csv \
    "$file" \
  	"$file".blocks \
  	/mnt/scratch/users/jps558/staging/hg38/hg38_TTAA.bed \
	  "$file"_p62_peaks.bed
  python /mnt/scratch/users/jps558/Stenning_Data_Analysis/BBPeakCaller_BRD4.py -p 9 -d 12500 \
    -i "$file"_p9_intermediate.csv \
    "$file" \
  	"$file".blocks \
  	/mnt/scratch/users/jps558/staging/hg38/hg38_TTAA.bed \
  	"$file"_p9_peaks.bed
  done
cd ../
  #Clean up block file ends

# Capture the part of the filename you want to keep
prefix1="ER_HyPB_EKDL230008858"
prefix2="ER_HyPB_HF552DSX7_L2_1"
prefix3="ER_HyPB_HFK3FDSX7_L2_1"
# Using 'mv' and 'sed' to rename the files
cd ER
for file in "${prefix1}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_nor.bed; do
    new_name=$(echo "$file" | sed "s/${prefix1}_sorted.qbed_filtered.qbed_/${prefix1}_/")
    mv "$file" "$new_name"
done
for file in "${prefix2}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_nor.bed; do
    new_name=$(echo "$file" | sed "s/${prefix2}_sorted.qbed_filtered.qbed_/${prefix2}_/")
    mv "$file" "$new_name"
done
for file in "${prefix3}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_nor.bed; do
    new_name=$(echo "$file" | sed "s/${prefix3}_sorted.qbed_filtered.qbed_/${prefix3}_/")
    mv "$file" "$new_name"
done
for file in "${prefix1}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_r.bed; do
    new_name=$(echo "$file" | sed "s/${prefix1}_sorted.qbed_filtered.qbed_/${prefix1}_/")
    mv "$file" "$new_name"
done
for file in "${prefix2}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_r.bed; do
    new_name=$(echo "$file" | sed "s/${prefix2}_sorted.qbed_filtered.qbed_/${prefix2}_/")
    mv "$file" "$new_name"
done
for file in "${prefix3}"*_sorted.qbed_filtered.qbed_fdr_bh_0.05_peaks_r.bed; do
    new_name=$(echo "$file" | sed "s/${prefix3}_sorted.qbed_filtered.qbed_/${prefix3}_/")
    mv "$file" "$new_name"
done
cd ../
# Capture the part of the filename you want to keep
prefix1="WT_HyPB_EKDL230008858"
prefix2="WT_HyPB_HF552DSX7_L2_1"
prefix3="WT_HyPB_HFK3FDSX7_L2_1"
# Using 'mv' and 'sed' to rename the files
cd WT
for file in "${prefix1}"*_sorted.qbed_filtered.qbed*.bed; do
    new_name=$(echo "$file" | sed "s/${prefix1}_sorted.qbed_filtered.qbed_/${prefix1}_/")
    mv "$file" "$new_name"
done
for file in "${prefix2}"*_sorted.qbed_filtered.qbed*.bed; do
    new_name=$(echo "$file" | sed "s/${prefix2}_sorted.qbed_filtered.qbed_/${prefix2}_/")
    mv "$file" "$new_name"
done
for file in "${prefix3}"*_sorted.qbed_filtered.qbed*.bed; do
    new_name=$(echo "$file" | sed "s/${prefix3}_sorted.qbed_filtered.qbed_/${prefix3}_/")
    mv "$file" "$new_name"
done
cd /mnt/scratch/users/jps558/Stenning_Data_Analysis
```


### Bedtools Window
#### Data found in '3 - called peak files Needed from Bedtools Window'

Once the peak files from my data were generated, I am then able to compare their overlap with existing ER ChIP-Seq data.

First we needed to aquire the data from a previous ChIP-Seq experiment of Andy's available from the GEO under GSE109820. In this experiment there were two types of files - .bed and .narrowPeaks - only the .bed files were relevant and as such they were sorted out from the remaining files.

```{bash get data, eval=FALSE}
mkdir 202306_JS_BulkCC/ChIP
cd 202306_JS_BulkCC/ChIP
curl -L -o GSE109820_RAW.tar https://ftp.ncbi.nlm.nih.gov/geo/series/GSE109nnn/GSE109820/suppl/GSE109820_RAW.tar
tar -xvf GSE109820_RAW.tar

#Separate and backup useful bed files from narrow peaks that were unused. 

mkdir narrow
mkdir bed
mkdir unused
mv *narrow* narrow/
mv *B45* unused/
mv *C45* unused/
mv *D45* unused/
mv *B0* unused/
mv *C0* unused/
mv *D0* unused/
cp *bed* bed/
gunzip *.gz
```

Everything was then compiled into a single directory before carrying out the window analysis. There are four different ESR1 ChIP-Seq peak files each with three time points - 0, 45 and 90 minutes post E2 treatment. Initially, I looked at using both the 45 and 90 minute files. However, it became apparent that there was not much difference between the two, with the 90 minute files showing consistently higher mapping. As a result, only the 90 minute files were carried forward in downstream analysis.

```{bash collect data, eval=FALSE}
mkdir /mnt/scratch/users/jps558/Stenning_Data_Analysis/202306_JS_BulkCC/outputs/window
cd /mnt/scratch/users/jps558/Stenning_Data_Analysis/202306_JS_BulkCC/ChIP
cp *GSM29704*90_peaks* ../outputs/window
cp *peaks.bed ../outputs/window
cd ../outputs


cd ../outputs/ER
for file in *peaks*; do
  echo "$file"
  cp "$file" ../window
done 
cd ../outputs/WT
for file in *peaks*; do
  echo "$file"
  cp "$file" ../window
done 
cd /mnt/scratch/users/jps558/Stenning_Data_Analysis/
```

From here, the bedtools window program was run to generate overlap analysis similar to that carried out on SP1 cards.

```{bash Window, eval=FALSE}
module load BEDTools/2.30.0-GCC-11.2.0

cd 202306_JS_BulkCC/outputs/WT
for file in *peaks*; do
  echo "$file"
  cp "$file" ../window
done 

cd ../window
mkdir results
#Analyse ER peaks
for CC in *r.bed; do
	for ChIP in *GSM29704*; do
		echo $CC
		echo $ChIP
		bedtools window -c -w 1000 -a $CC -b $ChIP > results/1kbp_A_"$CC"_B_"$ChIP"
		echo "Next!"
		done
	done
#Analyse WT peaks
for CC in *p*_peaks.bed; do
	for ChIP in *GSM29704*; do
		echo $CC
		echo $ChIP
		bedtools window -w 1000 -c -a $CC -b $ChIP > results/1kbp_A_"$CC"_B_"$ChIP"
		echo "Next!"
		done
	done
	
#Genomic TTAA
#NOTE - this needed to be re-run as there was not enough disk space for output, script was unchanged

cp /mnt/scratch/users/jps558/staging/hg38/hg38_TTAA.bed /mnt/scratch/users/jps558/Stenning_Data_Analysis/202306_JS_BulkCC/outputs/window
for TTAA in *TTAA.bed; do
	for ChIP in *GSM29704*; do
		echo $TTAA
		echo $ChIP
		bedtools window -w 1000 -c -a $TTAA -b $ChIP > results/1kbp_A_"$TTAA"_B_"$ChIP"
		echo "Next!"
		done
	done
```

Then, to assess whether - given an increasing window up to 100,000 bp - the overlap would eventually approach 100%, additional code was run to generate outputs up to this distance. This was not done with genomic TTAA to avoid filling my hard drive.

```{bash increasing window, eval=FALSE}
module load BEDTools/2.30.0-GCC-11.2.0
cd 202306_JS_BulkCC/outputs/window
mkdir results/increasingwindow

#ER
for CC in *r.bed; do
	for ChIP in *GSM29704*; do
		echo $CC
		echo $ChIP
		mkdir results/increasingwindow/A_"$CC"_B_"$ChIP"
		bedtools window -c -w 1000 -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/1000bp_A_"$CC"_B_"$ChIP"
		bedtools window -c -w 5000 -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/5000bp_A_"$CC"_B_"$ChIP"
		i=10000
		while [ $i -le 100000 ]
		do
		echo $i
    		bedtools window -c -w $i -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/"$i"bp_A_"$CC"_B_"$ChIP"
    i=$(($i+10000))
done
done
done
#Undirected
for CC in *p*_peaks.bed; do
	for ChIP in *GSM29704*; do
		echo $CC
		echo $ChIP
		mkdir results/increasingwindow/A_"$CC"_B_"$ChIP"
				bedtools window -c -w 1000 -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/1000bp_A_"$CC"_B_"$ChIP"
		bedtools window -c -w 5000 -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/5000bp_A_"$CC"_B_"$ChIP"
		i=10000
		while [ $i -le 100000 ]
		do
		echo $i
    		bedtools window -c -w $i -a $CC -b $ChIP > results/increasingwindow/A_"$CC"_B_"$ChIP"/"$i"bp_A_"$CC"_B_"$ChIP"
    i=$(($i+10000))
done
done
done
```

These outputs were then transferred to my local hard drive to process in Rstudio.

## Working in Rstudio
#### Data found in '4 - Window outputs to plot in R'

### Overlap within 1000 bp

This analysis was carried out after analysing the SP1-HyPB calling card data from Moudgil et al., 2020, so the code used was largely taken from there. I started by loading the necessary packages and reading the data into tables.

```{r load data, eval=FALSE}

#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',
                  "Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)


```

Once the data is loaded, the number of peaks are identified and the number of peaks with no overlap are counted. These can then be used to determine the percentage of peaks that don't overlap with ChIP-Seq peaks, which then allows you to derive the percentage of peaks that do overlap.

```{r process data, eval=FALSE}
#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_A90)
HyPB_total=nrow(Undir_HyPB_A90)
TTAA_total=nrow(TTAA_A90)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

```

Finally, the data can be put into a table and a graph plotted to show the percentage overlap for each of the samples within 1000 bp of ESR1 ChIP-Seq peaks

```{r plot data, eval=FALSE}
#Create table to plot
A90_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
A90_Figure <- ggplot(data = A90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  
                        geom_col(color = "black", show.legend = FALSE) +
                        ylim(0, 100) + 
                        labs(x = "Name and Distance from ChIP-Seq Peak",
                             y = "% of CC peaks overlapping with ChIP-Seq peak",
                             title = "Calling Cards and ChIP-Seq Overlap",
                             subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HF552DSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample A90") + 
                        theme_dark() + 
                        coord_cartesian(expand = FALSE, clip = "off") + 
                        geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 


A90_Figure

```

This was repeated for each of the calling card peak files

### Overlap within an increasing window size up to 100,000 bp

The overall structure of the code to plot the increasing window outputs is the same as the one above, there are just many more input files and downstream variables to handle. First the data is loaded into tables.

```{r get increasing data, eval=FALSE}
#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
#ESR1
ESR1_HyPB_HF5_A90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/1000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/5000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/10000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/20000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/30000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/40000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/50000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/60000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/70000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/80000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/90000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/100000bp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

##Undirected HyPB
HyPB_HF5_A90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/1000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/5000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/10000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/20000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/30000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/40000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/50000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/60000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/70000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/80000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/90000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/100000bp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

```

Then, the total number of peaks and number of overlapping peaks can be determined, this is then used to calulate the percentage overlap as before.

```{r increasing percentage, eval=FALSE}
#Get total peaks
ESR1_HF5_total=nrow(ESR1_HyPB_HF5_A90_1000bp)
HyPB_HF5_total=nrow(HyPB_HF5_A90_1000bp)
#TTAA_HF5_total=nrow(TTAA_A90)

#Get overlapping peaks
#ESR1
ESR1_HF5_overlap_1000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_5000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_10000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_20000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_30000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_40000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_50000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_60000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_70000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_80000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_90000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_100000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#HyPB
HyPB_HF5_overlap_1000bp = as.numeric(length(which(HyPB_HF5_A90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_5000bp = as.numeric(length(which(HyPB_HF5_A90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_10000bp = as.numeric(length(which(HyPB_HF5_A90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_20000bp = as.numeric(length(which(HyPB_HF5_A90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_30000bp = as.numeric(length(which(HyPB_HF5_A90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_40000bp = as.numeric(length(which(HyPB_HF5_A90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_50000bp = as.numeric(length(which(HyPB_HF5_A90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_60000bp = as.numeric(length(which(HyPB_HF5_A90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_70000bp = as.numeric(length(which(HyPB_HF5_A90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_80000bp = as.numeric(length(which(HyPB_HF5_A90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_90000bp = as.numeric(length(which(HyPB_HF5_A90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_100000bp = as.numeric(length(which(HyPB_HF5_A90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
#ESR1
Percentage_ESR1_HF5_1000bp = (1-(ESR1_HF5_overlap_1000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_5000bp = (1-(ESR1_HF5_overlap_5000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_10000bp = (1-(ESR1_HF5_overlap_10000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_20000bp = (1-(ESR1_HF5_overlap_20000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_30000bp = (1-(ESR1_HF5_overlap_30000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_40000bp = (1-(ESR1_HF5_overlap_40000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_50000bp = (1-(ESR1_HF5_overlap_50000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_60000bp = (1-(ESR1_HF5_overlap_60000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_70000bp = (1-(ESR1_HF5_overlap_70000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_80000bp = (1-(ESR1_HF5_overlap_80000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_90000bp = (1-(ESR1_HF5_overlap_90000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_100000bp = (1-(ESR1_HF5_overlap_100000bp/ESR1_HF5_total))*100

#HyPB
Percentage_HyPB_HF5_1000bp = (1-(HyPB_HF5_overlap_1000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_5000bp = (1-(HyPB_HF5_overlap_5000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_10000bp = (1-(HyPB_HF5_overlap_10000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_20000bp = (1-(HyPB_HF5_overlap_20000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_30000bp = (1-(HyPB_HF5_overlap_30000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_40000bp = (1-(HyPB_HF5_overlap_40000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_50000bp = (1-(HyPB_HF5_overlap_50000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_60000bp = (1-(HyPB_HF5_overlap_60000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_70000bp = (1-(HyPB_HF5_overlap_70000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_80000bp = (1-(HyPB_HF5_overlap_80000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_90000bp = (1-(HyPB_HF5_overlap_90000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_100000bp = (1-(HyPB_HF5_overlap_100000bp/HyPB_HF5_total))*100

```

Finally, the plot can be made by generating a table and using that to plot the data in ggplot.

```{r plot increasing, eval=FALSE}
#Percetage table
graphorder= c('E1', 
              'E5',
              'E10',
              'E20',
              'E30',
              'E40',
              'E50',
              'E60',
              'E70',
              'E80',
              'E90',
              'E100',
              'H1', 
              'H5',
              'H10',
              'H20',
              'H30',
              'H40',
              'H50',
              'H60',
              'H70',
              'H80',
              'H90',
              'H100')
HF5_A90_Incrw_Finalpercentage_table = data.frame(Name=graphorder,
                                                 Total=rep(c(ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total)),
                                                 Number_Not_Overlapping_withChIP= c(ESR1_HF5_overlap_1000bp,
                                                                     ESR1_HF5_overlap_5000bp,
                                                                     ESR1_HF5_overlap_10000bp,
                                                                     ESR1_HF5_overlap_20000bp,
                                                                     ESR1_HF5_overlap_30000bp,
                                                                     ESR1_HF5_overlap_40000bp,
                                                                     ESR1_HF5_overlap_50000bp,
                                                                     ESR1_HF5_overlap_60000bp,
                                                                     ESR1_HF5_overlap_70000bp,
                                                                     ESR1_HF5_overlap_80000bp,
                                                                     ESR1_HF5_overlap_90000bp,
                                                                     ESR1_HF5_overlap_100000bp,
                                                                     HyPB_HF5_overlap_1000bp,
                                                                     HyPB_HF5_overlap_5000bp,
                                                                     HyPB_HF5_overlap_10000bp,
                                                                     HyPB_HF5_overlap_20000bp,
                                                                     HyPB_HF5_overlap_30000bp,
                                                                     HyPB_HF5_overlap_40000bp,
                                                                     HyPB_HF5_overlap_50000bp,
                                                                     HyPB_HF5_overlap_60000bp,
                                                                     HyPB_HF5_overlap_70000bp,
                                                                     HyPB_HF5_overlap_80000bp,
                                                                     HyPB_HF5_overlap_90000bp,
                                                                     HyPB_HF5_overlap_100000bp),
                                                 PercentageOverlapping= c(
                                                   Percentage_ESR1_HF5_1000bp,
                                                   Percentage_ESR1_HF5_5000bp,
                                                   Percentage_ESR1_HF5_10000bp,
                                                   Percentage_ESR1_HF5_20000bp,
                                                   Percentage_ESR1_HF5_30000bp,
                                                   Percentage_ESR1_HF5_40000bp,
                                                   Percentage_ESR1_HF5_50000bp,
                                                   Percentage_ESR1_HF5_60000bp,
                                                   Percentage_ESR1_HF5_70000bp,
                                                   Percentage_ESR1_HF5_80000bp,
                                                   Percentage_ESR1_HF5_90000bp,
                                                   Percentage_ESR1_HF5_100000bp,
                                                   Percentage_HyPB_HF5_1000bp,
                                                   Percentage_HyPB_HF5_5000bp,
                                                   Percentage_HyPB_HF5_10000bp,
                                                   Percentage_HyPB_HF5_20000bp,
                                                   Percentage_HyPB_HF5_30000bp,
                                                   Percentage_HyPB_HF5_40000bp,
                                                   Percentage_HyPB_HF5_50000bp,
                                                   Percentage_HyPB_HF5_60000bp,
                                                   Percentage_HyPB_HF5_70000bp,
                                                   Percentage_HyPB_HF5_80000bp,
                                                   Percentage_HyPB_HF5_90000bp,
                                                   Percentage_HyPB_HF5_100000bp))

#Plot
HF5_A90_Incrw_Figure <- ggplot(data = HF5_A90_Incrw_Finalpercentage_table, 
                               aes(x= factor(Name, graphorder), 
                                   y = PercentageOverlapping,
                                   fill = Name, 
                                   stat("identity"))) +  
                                    geom_col(color = "black",
                                              show.legend = FALSE) +  
                                    ylim(0, 100) + 
                                    labs(x = "Distance from ChIP-Seq Peak (kbp)",                                                                       y = "% of CC peaks overlapping with ChIP-Seq peak",                                                            title = "Calling Cards and ChIP-Seq Overlap",                                                                 subtitle = "ESR1-HyPB and HyPB sample HF552DSX7 overlap with ESR1 ChIP-Seq \npeaks from sample A90, increasing the distance window from 1kbp to 100kbp") +  
                                    theme_dark() + 
                                    coord_cartesian(expand = FALSE, clip = "off") + 
                                    geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 


HF5_A90_Incrw_Figure

```

# Results

## What is this data

Firstly, the data was provided to us as fastq files, but each replicate appears to have four files. For ER replicate one, two are labelled ER1_EKDL230008858-1A_HF552DSX7\* and the other two are ER1_EKDL230008858-1A_HFK3FDSX7\*. Each of these has either of these two suffixes L2_1.fq or L2_2.fq. Through looking at the FastQC data, it appeared that L2_2.fq files had a much more varied sequence at the beginning of each read, where as the vast majority of L2_1.fq had the same beginning sequence, that matched with the appropriate barcode for that condition.

With this info, I determined that the L2_2.fq files were the paired ended read, this can be discarded in bulk CC sequencing and so only the L2_1.fq files were carried forward with the analysis. They were treated as separate files throughout this process as I was unsure why there were two - it has now been pointed out to me, this is likely due to the fact they were run in two different flow cells. As such, they could have been combined. Since writing this document, further analysis combining these files and processing them in the same way found there was no improvement to the data, in fact it appears to worsen the result - this will be discussed later.

For the initial analysis, I found that the number of overlapping reads was slightly better using peaks from [X]\_EKDL230008858-1A_HFK3FDSX7 files than [X]\_EKDL230008858-1A_HF552DSX7. As such, I will only be discussing data from [X]\_EKDL230008858-1A_HFK3FDSX7_L2_1 files in this markdown.

The ChIP-Seq data from GSE109820 contained four replicates, A, B, C and D. Each of these replicates had samples collected at 0, 45 and 90 mintues post-treament with E2 (an ER agonist). The 0 time sample was discarded and the difference between 45 min and 90 min samples were tested by looking at replicate A.

## Calling cards with 1000 bp overlap window with one ChIP-Seq replicate - A45 and A90

### Figure 1

```{r ER_A45, echo=FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
install.packages("ggthemes", repos = "http://cran.us.r-project.org")

#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_A45 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970403_A45_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_A45 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970403_A45_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_A45 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970403_A45_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_A45)
HyPB_total=nrow(Undir_HyPB_A45)
TTAA_total=nrow(TTAA_A45)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_A45$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_A45$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_A45$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

#Create table to plot
A45_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
A45_Figure <- ggplot(data = A45_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HFK3FDSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample A45") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(A45_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/HFK3FDSX7_A45.csv", row.names=FALSE)
A45_Figure
########################################################################

```

### Figure 2

```{r ER_A90, echo= FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes", repos = "http://cran.us.r-project.org")

#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_A90)
HyPB_total=nrow(Undir_HyPB_A90)
TTAA_total=nrow(TTAA_A90)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

#Create table to plot
A90_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
A90_Figure <- ggplot(data = A90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HFK3FDSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample A90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(A90_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/HFK3FDSX7_A90.csv", row.names=FALSE)
A90_Figure
########################################################################


```

As you can see, there is both very little difference in the number of overlapping reads and very little difference between the overlap in A45 and A90. This was confirmed in the other replicates and as such the 45 minute samples will not be discussed further.

I found it surprising that so few HyPB-ER peaks overlapped with ChIP-Seq peaks and that the undirected calling cards had more overlap than ER directed, so I plotted the results from the remaining ChIP-Seq samples to see how they look.

## Calling cards with 1000 bp overlap window with one ChIP-Seq replicate - B90, C90, D90

### Figure 3

```{r ER_B90, echo = FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes", repos = "http://cran.us.r-project.org")

#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_B90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970407_B90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_B90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970407_B90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_B90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970407_B90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_B90)
HyPB_total=nrow(Undir_HyPB_B90)
TTAA_total=nrow(TTAA_B90)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_B90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_B90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_B90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

#Create table to plot
B90_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
B90_Figure <- ggplot(data = B90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HFK3FDSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample B90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(B90_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/HFK3FDSX7_B90.csv", row.names=FALSE)
B90_Figure
########################################################################

```

### Figure 4

```{r ER_C90, echo=FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes", repos = "http://cran.us.r-project.org")

#Load software
library(ggplot2)
library(ggthemes)


#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_C90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_C90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_C90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_C90)
HyPB_total=nrow(Undir_HyPB_C90)
TTAA_total=nrow(TTAA_C90)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_C90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_C90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_C90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

#Create table to plot
C90_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
C90_Figure <- ggplot(data = C90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HFK3FDSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample C90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(C90_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/HFK3FDSX7_C90.csv", row.names=FALSE)
C90_Figure
########################################################################


```

### Figure 5

```{r ER_D90, echo= FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes", repos = "http://cran.us.r-project.org")

#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_HyPB_D90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970413_D90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HyPB_D90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970413_D90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_D90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970413_D90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

#Counting the total reads
ESR1_total=nrow(ESR1_HyPB_D90)
HyPB_total=nrow(Undir_HyPB_D90)
TTAA_total=nrow(TTAA_D90)

#Count the number of reads with no overlap
ESR1_overlap = as.numeric(length(which(ESR1_HyPB_D90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_overlap = as.numeric(length(which(Undir_HyPB_D90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_D90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1 = (1-(ESR1_overlap/ESR1_total))*100
Percentage_HyPB = (1-(HyPB_overlap/HyPB_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

#Create table to plot
D90_Finalpercentage_table = data.frame(Name=c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp'),
                                       Total=rep(c(ESR1_total, HyPB_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_overlap, HyPB_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1, Percentage_HyPB, Percentage_TTAA))

#Then we plot
graphorder= c('ESR1-HyPB 1kbp', 'Undirected HyPB 1kbp', 'Genomic TTAA 1kbp')
D90_Figure <- ggplot(data = D90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \nsample HFK3FDSX7 within 1000 bp of ESR1 ChIP-Seq peaks from \nsample D90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(D90_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/HFK3FDSX7_D90.csv", row.names=FALSE)
D90_Figure
########################################################################


```

Unfortunately, it appears that all CC overlap with these ChIP-Seq peaks is also very low and undirected peaks are overlapping much more than directed. This was the same when using the peak files provided in the Moudgil et al., 2020.

## Increasing Calling Cards window up to 100,000 bp to assess overlap with ChIP-Seq peaks - A90 and C90

To see if eventually overlap would be saturated if the window size increases, I wrote a script to increase the window size up to 100,000 bp. In theory, as the window gets larger and larger, eventually all of the reads should overlap with a ChIP-Seq peak.

ChIP-Seq samples A90 and C90 were used as they represent the samples with the most and least overlap with CC peaks respectively.

### Figure 6

```{r ER Increasing window A90, echo= FALSE, message=FALSE, warning=FALSE}
#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")


##Read data
#Variables named <Window '-a'><Window'-b'>
#ESR1
# Initial code with 'Z' replaced by '1000'
ESR1_HyPB_HF5_A90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/1000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/5000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/10000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/20000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/30000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/40000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/50000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/60000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/70000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/80000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/90000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_A90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed/100000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

##Undirected HyPB
HyPB_HF5_A90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/1000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/5000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/10000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/20000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/30000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/40000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/50000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/60000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/70000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/80000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/90000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_A90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed/100000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

#Genomic TTAA
#Not run yet to save space

#Get total peaks
ESR1_HF5_total=nrow(ESR1_HyPB_HF5_A90_1000bp)
HyPB_HF5_total=nrow(HyPB_HF5_A90_1000bp)
#TTAA_HF5_total=nrow(TTAA_A90)

#Get overlapping peaks
#ESR1
ESR1_HF5_overlap_1000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_5000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_10000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_20000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_30000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_40000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_50000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_60000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_70000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_80000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_90000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_100000bp = as.numeric(length(which(ESR1_HyPB_HF5_A90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#HyPB
HyPB_HF5_overlap_1000bp = as.numeric(length(which(HyPB_HF5_A90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_5000bp = as.numeric(length(which(HyPB_HF5_A90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_10000bp = as.numeric(length(which(HyPB_HF5_A90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_20000bp = as.numeric(length(which(HyPB_HF5_A90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_30000bp = as.numeric(length(which(HyPB_HF5_A90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_40000bp = as.numeric(length(which(HyPB_HF5_A90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_50000bp = as.numeric(length(which(HyPB_HF5_A90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_60000bp = as.numeric(length(which(HyPB_HF5_A90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_70000bp = as.numeric(length(which(HyPB_HF5_A90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_80000bp = as.numeric(length(which(HyPB_HF5_A90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_90000bp = as.numeric(length(which(HyPB_HF5_A90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_100000bp = as.numeric(length(which(HyPB_HF5_A90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
#ESR1
Percentage_ESR1_HF5_1000bp = (1-(ESR1_HF5_overlap_1000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_5000bp = (1-(ESR1_HF5_overlap_5000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_10000bp = (1-(ESR1_HF5_overlap_10000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_20000bp = (1-(ESR1_HF5_overlap_20000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_30000bp = (1-(ESR1_HF5_overlap_30000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_40000bp = (1-(ESR1_HF5_overlap_40000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_50000bp = (1-(ESR1_HF5_overlap_50000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_60000bp = (1-(ESR1_HF5_overlap_60000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_70000bp = (1-(ESR1_HF5_overlap_70000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_80000bp = (1-(ESR1_HF5_overlap_80000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_90000bp = (1-(ESR1_HF5_overlap_90000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_100000bp = (1-(ESR1_HF5_overlap_100000bp/ESR1_HF5_total))*100

#HyPB
Percentage_HyPB_HF5_1000bp = (1-(HyPB_HF5_overlap_1000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_5000bp = (1-(HyPB_HF5_overlap_5000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_10000bp = (1-(HyPB_HF5_overlap_10000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_20000bp = (1-(HyPB_HF5_overlap_20000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_30000bp = (1-(HyPB_HF5_overlap_30000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_40000bp = (1-(HyPB_HF5_overlap_40000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_50000bp = (1-(HyPB_HF5_overlap_50000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_60000bp = (1-(HyPB_HF5_overlap_60000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_70000bp = (1-(HyPB_HF5_overlap_70000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_80000bp = (1-(HyPB_HF5_overlap_80000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_90000bp = (1-(HyPB_HF5_overlap_90000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_100000bp = (1-(HyPB_HF5_overlap_100000bp/HyPB_HF5_total))*100

#Percetage table
graphorder= c('E1', 
              'E5',
              'E10',
              'E20',
              'E30',
              'E40',
              'E50',
              'E60',
              'E70',
              'E80',
              'E90',
              'E100',
              'H1', 
              'H5',
              'H10',
              'H20',
              'H30',
              'H40',
              'H50',
              'H60',
              'H70',
              'H80',
              'H90',
              'H100')
HF5_A90_Incrw_Finalpercentage_table = data.frame(Name=graphorder,
                                                 Total=rep(c(ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total)),
                                                 Number_Not_Overlapping_withChIP= c(ESR1_HF5_overlap_1000bp,
                                                                     ESR1_HF5_overlap_5000bp,
                                                                     ESR1_HF5_overlap_10000bp,
                                                                     ESR1_HF5_overlap_20000bp,
                                                                     ESR1_HF5_overlap_30000bp,
                                                                     ESR1_HF5_overlap_40000bp,
                                                                     ESR1_HF5_overlap_50000bp,
                                                                     ESR1_HF5_overlap_60000bp,
                                                                     ESR1_HF5_overlap_70000bp,
                                                                     ESR1_HF5_overlap_80000bp,
                                                                     ESR1_HF5_overlap_90000bp,
                                                                     ESR1_HF5_overlap_100000bp,
                                                                     HyPB_HF5_overlap_1000bp,
                                                                     HyPB_HF5_overlap_5000bp,
                                                                     HyPB_HF5_overlap_10000bp,
                                                                     HyPB_HF5_overlap_20000bp,
                                                                     HyPB_HF5_overlap_30000bp,
                                                                     HyPB_HF5_overlap_40000bp,
                                                                     HyPB_HF5_overlap_50000bp,
                                                                     HyPB_HF5_overlap_60000bp,
                                                                     HyPB_HF5_overlap_70000bp,
                                                                     HyPB_HF5_overlap_80000bp,
                                                                     HyPB_HF5_overlap_90000bp,
                                                                     HyPB_HF5_overlap_100000bp),
                                                 PercentageOverlapping= c(
                                                   Percentage_ESR1_HF5_1000bp,
                                                   Percentage_ESR1_HF5_5000bp,
                                                   Percentage_ESR1_HF5_10000bp,
                                                   Percentage_ESR1_HF5_20000bp,
                                                   Percentage_ESR1_HF5_30000bp,
                                                   Percentage_ESR1_HF5_40000bp,
                                                   Percentage_ESR1_HF5_50000bp,
                                                   Percentage_ESR1_HF5_60000bp,
                                                   Percentage_ESR1_HF5_70000bp,
                                                   Percentage_ESR1_HF5_80000bp,
                                                   Percentage_ESR1_HF5_90000bp,
                                                   Percentage_ESR1_HF5_100000bp,
                                                   Percentage_HyPB_HF5_1000bp,
                                                   Percentage_HyPB_HF5_5000bp,
                                                   Percentage_HyPB_HF5_10000bp,
                                                   Percentage_HyPB_HF5_20000bp,
                                                   Percentage_HyPB_HF5_30000bp,
                                                   Percentage_HyPB_HF5_40000bp,
                                                   Percentage_HyPB_HF5_50000bp,
                                                   Percentage_HyPB_HF5_60000bp,
                                                   Percentage_HyPB_HF5_70000bp,
                                                   Percentage_HyPB_HF5_80000bp,
                                                   Percentage_HyPB_HF5_90000bp,
                                                   Percentage_HyPB_HF5_100000bp))

#WhenTTAA needed
#'Genomic TTAA 1kbp',
#'Genomic TTAA 5kbp',
#'Genomic TTAA 10kbp',
#'Genomic TTAA 15kbp',
#'Genomic TTAA 20kbp'
#TTAA_total,
#TTAA_total,
#TTAA_total,
#TTAA_total,
#TTAA_total,

#Plot
HF5_A90_Incrw_Figure <- ggplot(data = HF5_A90_Incrw_Finalpercentage_table, 
                               aes(x= factor(Name, graphorder), 
                                   y = PercentageOverlapping,
                                   fill = Name, 
                                   stat("identity"))) +  geom_col(color = "black",
                                                                  show.legend = FALSE) +  ylim(0, 100) + labs(x = "Distance from ChIP-Seq Peak (kbp)", 
                                                                                                              y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                              title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                              subtitle = "ESR1-HyPB and HyPB sample HFK3FDSX7 overlap with ESR1 ChIP-Seq \npeaks from sample A90, increasing the distance window from 1kbp to 100kbp") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(HF5_A90_Incrw_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/A90_increasingwindow.csv", row.names=FALSE)
HF5_A90_Incrw_Figure
########################################################################

```

### Figure 7

```{r ER increasing window C90, echo= FALSE, message=FALSE, warning=FALSE}
#Load software
library(ggplot2)
library(ggthemes)

#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")


##Read data
#Variables named <Window '-a'><Window'-b'>
#ESR1
# Initial code with 'Z' replaced by '1000'
ESR1_HyPB_HF5_C90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/1000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/5000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/10000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/20000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/30000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/40000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/50000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/60000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/70000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/80000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/90000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HyPB_HF5_C90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed/100000bp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_nor.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

##Undirected HyPB
HyPB_HF5_C90_1000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/1000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_5000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/5000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_10000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/10000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_20000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/20000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_30000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/30000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_40000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/40000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_50000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/50000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_60000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/60000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_70000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/70000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_80000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/80000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_90000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/90000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

HyPB_HF5_C90_100000bp = read.table("202306_JS_BulkCC/outputs/window/results/increasingwindow/A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed/100000bp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970410_C90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

#Genomic TTAA
#Not run yet to save space

#Get total peaks
ESR1_HF5_total=nrow(ESR1_HyPB_HF5_C90_1000bp)
HyPB_HF5_total=nrow(HyPB_HF5_C90_1000bp)
#TTAA_HF5_total=nrow(TTAA_C90)

#Get overlapping peaks
#ESR1
ESR1_HF5_overlap_1000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_5000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_10000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_20000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_30000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_40000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_50000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_60000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_70000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_80000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_90000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
ESR1_HF5_overlap_100000bp = as.numeric(length(which(ESR1_HyPB_HF5_C90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#HyPB
HyPB_HF5_overlap_1000bp = as.numeric(length(which(HyPB_HF5_C90_1000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_5000bp = as.numeric(length(which(HyPB_HF5_C90_5000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_10000bp = as.numeric(length(which(HyPB_HF5_C90_10000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_20000bp = as.numeric(length(which(HyPB_HF5_C90_20000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_30000bp = as.numeric(length(which(HyPB_HF5_C90_30000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_40000bp = as.numeric(length(which(HyPB_HF5_C90_40000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_50000bp = as.numeric(length(which(HyPB_HF5_C90_50000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_60000bp = as.numeric(length(which(HyPB_HF5_C90_60000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_70000bp = as.numeric(length(which(HyPB_HF5_C90_70000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_80000bp = as.numeric(length(which(HyPB_HF5_C90_80000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_90000bp = as.numeric(length(which(HyPB_HF5_C90_90000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap_100000bp = as.numeric(length(which(HyPB_HF5_C90_100000bp$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
#ESR1
Percentage_ESR1_HF5_1000bp = (1-(ESR1_HF5_overlap_1000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_5000bp = (1-(ESR1_HF5_overlap_5000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_10000bp = (1-(ESR1_HF5_overlap_10000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_20000bp = (1-(ESR1_HF5_overlap_20000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_30000bp = (1-(ESR1_HF5_overlap_30000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_40000bp = (1-(ESR1_HF5_overlap_40000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_50000bp = (1-(ESR1_HF5_overlap_50000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_60000bp = (1-(ESR1_HF5_overlap_60000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_70000bp = (1-(ESR1_HF5_overlap_70000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_80000bp = (1-(ESR1_HF5_overlap_80000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_90000bp = (1-(ESR1_HF5_overlap_90000bp/ESR1_HF5_total))*100
Percentage_ESR1_HF5_100000bp = (1-(ESR1_HF5_overlap_100000bp/ESR1_HF5_total))*100

#HyPB
Percentage_HyPB_HF5_1000bp = (1-(HyPB_HF5_overlap_1000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_5000bp = (1-(HyPB_HF5_overlap_5000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_10000bp = (1-(HyPB_HF5_overlap_10000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_20000bp = (1-(HyPB_HF5_overlap_20000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_30000bp = (1-(HyPB_HF5_overlap_30000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_40000bp = (1-(HyPB_HF5_overlap_40000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_50000bp = (1-(HyPB_HF5_overlap_50000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_60000bp = (1-(HyPB_HF5_overlap_60000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_70000bp = (1-(HyPB_HF5_overlap_70000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_80000bp = (1-(HyPB_HF5_overlap_80000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_90000bp = (1-(HyPB_HF5_overlap_90000bp/HyPB_HF5_total))*100
Percentage_HyPB_HF5_100000bp = (1-(HyPB_HF5_overlap_100000bp/HyPB_HF5_total))*100

#Percetage table
graphorder= c('E1', 
              'E5',
              'E10',
              'E20',
              'E30',
              'E40',
              'E50',
              'E60',
              'E70',
              'E80',
              'E90',
              'E100',
              'H1', 
              'H5',
              'H10',
              'H20',
              'H30',
              'H40',
              'H50',
              'H60',
              'H70',
              'H80',
              'H90',
              'H100')
HF5_C90_Incrw_Finalpercentage_table = data.frame(Name=graphorder,
                                                 Total=rep(c(ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             ESR1_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total, 
                                                             HyPB_total,
                                                             HyPB_total, 
                                                             HyPB_total)),
                                                 Number_Not_Overlapping_withChIP= c(ESR1_HF5_overlap_1000bp,
                                                                     ESR1_HF5_overlap_5000bp,
                                                                     ESR1_HF5_overlap_10000bp,
                                                                     ESR1_HF5_overlap_20000bp,
                                                                     ESR1_HF5_overlap_30000bp,
                                                                     ESR1_HF5_overlap_40000bp,
                                                                     ESR1_HF5_overlap_50000bp,
                                                                     ESR1_HF5_overlap_60000bp,
                                                                     ESR1_HF5_overlap_70000bp,
                                                                     ESR1_HF5_overlap_80000bp,
                                                                     ESR1_HF5_overlap_90000bp,
                                                                     ESR1_HF5_overlap_100000bp,
                                                                     HyPB_HF5_overlap_1000bp,
                                                                     HyPB_HF5_overlap_5000bp,
                                                                     HyPB_HF5_overlap_10000bp,
                                                                     HyPB_HF5_overlap_20000bp,
                                                                     HyPB_HF5_overlap_30000bp,
                                                                     HyPB_HF5_overlap_40000bp,
                                                                     HyPB_HF5_overlap_50000bp,
                                                                     HyPB_HF5_overlap_60000bp,
                                                                     HyPB_HF5_overlap_70000bp,
                                                                     HyPB_HF5_overlap_80000bp,
                                                                     HyPB_HF5_overlap_90000bp,
                                                                     HyPB_HF5_overlap_100000bp),
                                                 PercentageOverlapping= c(
                                                   Percentage_ESR1_HF5_1000bp,
                                                   Percentage_ESR1_HF5_5000bp,
                                                   Percentage_ESR1_HF5_10000bp,
                                                   Percentage_ESR1_HF5_20000bp,
                                                   Percentage_ESR1_HF5_30000bp,
                                                   Percentage_ESR1_HF5_40000bp,
                                                   Percentage_ESR1_HF5_50000bp,
                                                   Percentage_ESR1_HF5_60000bp,
                                                   Percentage_ESR1_HF5_70000bp,
                                                   Percentage_ESR1_HF5_80000bp,
                                                   Percentage_ESR1_HF5_90000bp,
                                                   Percentage_ESR1_HF5_100000bp,
                                                   Percentage_HyPB_HF5_1000bp,
                                                   Percentage_HyPB_HF5_5000bp,
                                                   Percentage_HyPB_HF5_10000bp,
                                                   Percentage_HyPB_HF5_20000bp,
                                                   Percentage_HyPB_HF5_30000bp,
                                                   Percentage_HyPB_HF5_40000bp,
                                                   Percentage_HyPB_HF5_50000bp,
                                                   Percentage_HyPB_HF5_60000bp,
                                                   Percentage_HyPB_HF5_70000bp,
                                                   Percentage_HyPB_HF5_80000bp,
                                                   Percentage_HyPB_HF5_90000bp,
                                                   Percentage_HyPB_HF5_100000bp))

#WhenTTAA needed
#'Genomic TTAA 1kbp',
#'Genomic TTAA 5kbp',
#'Genomic TTAA 10kbp',
#'Genomic TTAA 15kbp',
#'Genomic TTAA 20kbp'
#TTAA_total,
#TTAA_total,
#TTAA_total,
#TTAA_total,
#TTAA_total,

#Plot
HF5_C90_Incrw_Figure <- ggplot(data = HF5_C90_Incrw_Finalpercentage_table, 
                               aes(x= factor(Name, graphorder), 
                                   y = PercentageOverlapping,
                                   fill = Name, 
                                   stat("identity"))) +  geom_col(color = "black",
                                                                  show.legend = FALSE) +  ylim(0, 100) + labs(x = "Distance from ChIP-Seq Peak (kbp)", 
                                                                                                              y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                              title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                              subtitle = "ESR1-HyPB and HyPB sample HFK3FDSX7 overlap with ESR1 ChIP-Seq \npeaks from sample C90, increasing the distance window from 1kbp to 100kbp") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed", linewidth = 1) 

write.csv(HF5_C90_Incrw_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/C90_increasingwindow.csv", row.names=FALSE)
HF5_C90_Incrw_Figure
########################################################################

```

Although these plots show a trend up to saturation, neither of them make it to 83% - the percentage found in Wang et al., 2012. This is concerning and I'm not sure what is causing both the lack over ESR1 directed CC overlap with ChIP-Seq and why the undirected calling cards are overlapping more than the directed.

## Supplimentary analysis after writing this document

Since gathering this evidence to send to you, I combined the data across the two flow cells for both samples to increase the statistical power of the data and used the '-r' refine flag in the peak calling script to try and improve the number of overlapping peaks. 

### Figure 8

```{r combined flow cell, echo= FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes")

#Load software
library(ggplot2)
library(ggthemes)

#Set working directory
#setwd("//userfs/jps558/w2k/Desktop/Local Calling Cards Analysis/Data/JPS/20231117RmdOutputs") 


#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_EK_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_EKDL230008858_fdr_bh_0.05_peaks_r.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_EK_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_EKDL230008858_p9_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
TTAA_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_hg38_TTAA.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = Genocol_names)

ESR1_HF5_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HF552DSX7_L2_1_fdr_bh_0.05_peaks_r.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HF5_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HF552DSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_HF3_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_HFK3FDSX7_L2_1_fdr_bh_0.05_peaks_r.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_HF3_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_HFK3FDSX7_L2_1_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

#Counting the total reads
ESR1_EK_total=nrow(ESR1_EK_HyPB_A90)
HyPB_EK_total=nrow(Undir_EK_HyPB_A90)
TTAA_total=nrow(TTAA_A90)

ESR1_HF5_total=nrow(ESR1_HF5_HyPB_A90)
HyPB_HF5_total=nrow(Undir_HF5_HyPB_A90)

ESR1_HF3_total=nrow(ESR1_HF3_HyPB_A90)
HyPB_HF3_total=nrow(Undir_HF3_HyPB_A90)
#Count the number of reads with no overlap
ESR1_EK_overlap = as.numeric(length(which(ESR1_EK_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_EK_overlap = as.numeric(length(which(Undir_EK_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
TTAA_overlap = as.numeric(length(which(TTAA_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

ESR1_HF5_overlap = as.numeric(length(which(ESR1_HF5_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF5_overlap = as.numeric(length(which(Undir_HF5_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

ESR1_HF3_overlap = as.numeric(length(which(ESR1_HF3_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_HF3_overlap = as.numeric(length(which(Undir_HF3_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
#Determine overlap percentage
Percentage_ESR1_EK = (1-(ESR1_EK_overlap/ESR1_EK_total))*100
Percentage_HyPB_EK = (1-(HyPB_EK_overlap/HyPB_EK_total))*100
Percentage_TTAA = (1-(TTAA_overlap/TTAA_total))*100

Percentage_ESR1_HF5 = (1-(ESR1_HF5_overlap/ESR1_HF5_total))*100
Percentage_HyPB_HF5 = (1-(HyPB_HF5_overlap/HyPB_HF5_total))*100

Percentage_ESR1_HF3 = (1-(ESR1_HF3_overlap/ESR1_HF3_total))*100
Percentage_HyPB_HF3 = (1-(HyPB_HF3_overlap/HyPB_HF3_total))*100
#Create table to plot
A90_Finalpercentage_table = data.frame(Name=c('ER_EK', 'HyPB_EK', 'TTAA_1', 'ER_HF5', 'HyPB_HF5', 'TTAA_2', 'ER_HF3', 'HyPB_HF3', 'TTAA_3'),
                                       Total=rep(c(ESR1_EK_total, HyPB_EK_total, TTAA_total, ESR1_HF5_total, HyPB_HF5_total, TTAA_total, ESR1_HF3_total, HyPB_HF3_total, TTAA_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_EK_overlap, HyPB_EK_overlap, TTAA_overlap, ESR1_HF5_overlap, HyPB_HF5_overlap, TTAA_overlap, ESR1_HF3_overlap, HyPB_HF3_overlap, TTAA_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1_EK, Percentage_HyPB_EK, Percentage_TTAA, Percentage_ESR1_HF5, Percentage_HyPB_HF5, Percentage_TTAA, Percentage_ESR1_HF3, Percentage_HyPB_HF3, Percentage_TTAA))

#Then we plot
graphorder= c('ER_EK', 'HyPB_EK', 'TTAA_1', 'ER_HF5', 'HyPB_HF5', 'TTAA_2', 'ER_HF3', 'HyPB_HF3', 'TTAA_3')
A90_Figure <- ggplot(data = A90_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlapping peaks from ESR1-HyPB and HyPB \n combined flow cell within 1000 bp of ESR1 ChIP-Seq peaks from sample A90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed") 

write.csv(A90_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/Comparing ESR1 combined flow cell r and nor.csv", row.names=FALSE)
A90_Figure
########################################################################

```

Unfortunately, this didn't only fail to improve the number of overlapping peaks, but it actually reduces the number of overlapping peaks. 

I couldn't understand how this was possible until I noticed something odd in the intermediate files generated by the BBPeakCaller_TF.py custom script on Arnhav's GitHub - https://github.com/arnavm/calling_cards. 

When the entries in the file are sorted by 'corrected_pValue' smallest to largest, it appears that any peak with a corrected Pvalue lower than 0.05 has been rejected. I would have expected the opposite to be true, and visualizing the peak with the lowest corrected Pvalue shows a region next to a commonly upregulated gene in breast cancer with sharp ESR1 CC peaks and almost no undirected insertion. 

Given this is what I expected to be generated from the peak calling script as a top peak, I am confused as to why this peak has been rejected. To investigate, I took all of the rejected peaks from the ESR1 intermediate file and peaks with similar Pvalues in the undirected calling cards (as the correction is not carried out by the BRD4 script) and determined their overlap with ChIP-Seq peaks.

### Figure 9
```{r rejectvsaccept, echo= FALSE, message=FALSE, warning=FALSE}
############# LOOK HERE ######################
#add or delete the '#' symbol below on install packages
#install.packages("Rtools")
#install.packages("ggthemes")

#Load software
library(ggplot2)
library(ggthemes)


#Set column names
col_names = c("Chromosome", "Position Start", "Position End", "Number of Overlaps with ChIP-Seq Peaks")
Genocol_names = c("Chromosome", "Position Start", "Position End", 'Name', 'Score', 'Strand',"Number of Overlaps with ChIP-Seq Peaks")

##Read data
#Variables named <Window '-a'><Window'-b'>
ESR1_EK_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_ER_HyPB_EKDL230008858_rejected_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_EK_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/1kbp_A_WT_HyPB_EKDL230008858_rejected_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

ESR1_accept_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/accepted/1kbp_A_ER_HyPB_EKDL230008858_fdr_bh_0.05_peaks_r.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)
Undir_Accept_HyPB_A90 = read.table("202306_JS_BulkCC/outputs/window/results/accepted/1kbp_A_WT_HyPB_EKDL230008858_p30_peaks.bed_B_GSM2970404_A90_peaks.bed", header = TRUE, sep = "\t", col.names = col_names)

#Counting the total reads
ESR1_A90_total=nrow(ESR1_EK_HyPB_A90)
HyPB_A90_total=nrow(Undir_EK_HyPB_A90)


ESR1_accept_total=nrow(ESR1_accept_HyPB_A90)
HyPB_accept_total=nrow(Undir_Accept_HyPB_A90)


#Count the number of reads with no overlap
ESR1_A90_overlap = as.numeric(length(which(ESR1_EK_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_A90_overlap = as.numeric(length(which(Undir_EK_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

ESR1_accept_overlap = as.numeric(length(which(ESR1_accept_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))
HyPB_accept_overlap = as.numeric(length(which(Undir_Accept_HyPB_A90$Number.of.Overlaps.with.ChIP.Seq.Peaks==0)))

#Determine overlap percentage
Percentage_ESR1_A90 = (1-(ESR1_A90_overlap/ESR1_A90_total))*100
Percentage_HyPB_A90 = (1-(HyPB_A90_overlap/HyPB_A90_total))*100

Percentage_ESR1_accept = (1-(ESR1_accept_overlap/ESR1_accept_total))*100
Percentage_HyPB_accept = (1-(HyPB_accept_overlap/HyPB_accept_total))*100


#Create table to plot
RejectVsAccept_Finalpercentage_table = data.frame(Name=c('ER_Reject', 'ER_Accept', 'HyPB_Reject','HyPB_Accept'),
                                       Total=rep(c(ESR1_A90_total, ESR1_accept_total, HyPB_A90_total, HyPB_accept_total)),
                                       Number_Not_Overlapping_withChIP= c(ESR1_A90_overlap, ESR1_accept_overlap, HyPB_A90_overlap, HyPB_accept_overlap),
                                       PercentageOverlapping= c(Percentage_ESR1_A90, Percentage_ESR1_accept, Percentage_HyPB_A90, Percentage_HyPB_accept))

#Then we plot
graphorder= c('ER_Reject', 'ER_Accept', 'HyPB_Reject','HyPB_Accept')
A90_Figure <- ggplot(data = RejectVsAccept_Finalpercentage_table, 
                     aes(x= factor(Name, graphorder), 
                         y = PercentageOverlapping,
                         fill = Name, 
                         stat("identity"))) +  geom_col(color = "black",
                                                        show.legend = FALSE) +  ylim(0, 100) + labs(x = "Name and Distance from ChIP-Seq Peak", 
                                                                                                    y = "% of CC peaks overlapping with ChIP-Seq peak",
                                                                                                    title = "Calling Cards and ChIP-Seq Overlap",
                                                                                                    subtitle = "Determining percentage of overlap from ESR1-HyPB and HyPB \n accepted and rejected peaks within 1000 bp of ESR1 ChIP-Seq peaks from sample A90") +  theme_dark() + coord_cartesian(expand = FALSE, clip = "off") + geom_hline(yintercept = 83, linetype = "dashed") 


A90_Figure
write.csv(RejectVsAccept_Finalpercentage_table, "/mnt/scratch/users/jps558/Stenning_Data_Analysis/stats/Comparing ESR1 rejected peaks to non-rejected.csv", row.names=FALSE)
########################################################################
```

This doesn't fix my data, but it does bring it more in line with what I was expecting, more ESR1 CC overlap with ChIP-seq than undirected. 

There are additional steps carried out by the BBPeakCaller_TF.py script that I have not captured by analyzing the data this way. I know this because there were ~20,000 lines that were not rejected, however only ~5,000 of these were written into the peak file. 

# Discussion

I had originally come up with the hypothesis that this apparent lack of overlap could be caused by the Bayesian block method of analysis. The undirected HyPB has an affinity for BRD4, shown by their co-immunoprecipitation, which means that it will be inserting SRTs into enhancer and super enhancer sites. ESR1 is thought to act as a mediator of enchancer and promoter interaction, meaning that both ESR1 and BRD4 could be co-occupying the same sites. This could be having an effect where the undirected insertions are clustered in very similar places to directed insertions, so that when the Baysean block are made, the peak calling program might not be able to correctly identify the difference between background and genuine signal - effectively masking or diminishing the signal. I believe this is supported by the fact that gene ontology shows that alot of the most represented genes in ER directed and undirected calling cards are the same.

While this hypothesis could still be correct - with the additional information regarding the rejected peaks appearing to have the lowest Pvalues - there appears to be more to the problem than BRD4-ESR1 co-localisation confounding peak calling. 

Could you please help me to understand why it appears that loci with strong ESR1 CC integration and almost no WT-HyPB CC integration have been rejected by your script? 


# Future directions

To see if BRD4 co-localisation is confounding peak calling, Andy has suggested to subtract a BRD4 ChIP-Seq bed file from my ER ChIP-Seq. This should, in theory, remove all of the enhancer sites that would be bound in these cells and if used to run the bedtools window analysis, it could be used to give an indication whether my hypothesis is correct. It could then be tested by trying to find alternative analysis methods.
